### PR TITLE
Sync upstream

### PR DESCRIPTION
**Pull request by andredestro**
### Motivation and Context

The OutSystems branch is very out of date to the upstream.
As the changes in the plugin for MABS 11 are breaking changes, it can be a good time to do this sync.

### Checklist

- [ ] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

**Review by markemer — DISMISSED** [review]

**Comment by markemer** [comment]
Wait this still has the `macos-11` stuff in it - the android stuff looks better, though.

**Comment by markemer** [comment]
Yeah, these commits appear to be missing: https://github.com/apache/cordova-plugin-statusbar/pull/271 - but they're merged upstream - hrm. Yeah, seems 4.0.0 is too old, we'll have to merge main. I think there are only three 3 PRs extra anyway.

**Review by markemer — CHANGES_REQUESTED** [review]
Yeah - we need to merge HEAD instead of 4.0.0, or cherry pick the new PRs to get CI to work.

File: src/android/StatusBar.java [hunk 1]
```diff
@@ -118,194 +133,115 @@ else if(Build.VERSION.SDK_INT == Build.VERSION_CODES.KITKAT) {
...
+            case ACTION_READY:
+                boolean statusBarVisible = (window.getAttributes().flags & WindowManager.LayoutParams.FLAG_FULLSCREEN) == 0;
+                callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK, statusBarVisible));
+                return true;
 
-        if("isStatusBarOverlayingWebview".equals(action)) {
-            boolean statusBarVisible = (window.getAttributes().flags & WindowManager.LayoutParams.FLAG_FULLSCREEN) == 0;
-            callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK, doOverlay && statusBarVisible));
-            return true;
-        }
```
Comment (jcesarmobile) [inline src/android/StatusBar.java:136]: This seems outsystems specific, should not be removed

Comment (jcesarmobile) [reply src/android/StatusBar.java:136]: nevermind, the code is still there, just missing the `isStatusBarOverlayingWebview` declaration

**Review by jcesarmobile — CHANGES_REQUESTED** [review]
`ACTION_IS_OVERLAYS_WEB_VIEW` and `ACTION_GET_HEIGHT` have been added to the switch statement, but they are not declared.

**Review by jcesarmobile — APPROVED** [review]

**Review by markemer — APPROVED** [review]

**Pull request merged**